### ISSUE TITLE
creating and merging Dicts with a value combiner

### ISSUE THREAD

**Issue by StefanKarpinski** [JuliaLang/julia#20699]
It would be handy to be able to do `merge!(+, d1, d2)` and such to combine Dicts with a combiner function. Similarly, writing `Dict(+, (k(x) => v(x) for x in v))` would be handy for construction.

**Comment by ararslan** [comment]
To clarify, is the combining function applied to elements which match between the two input dictionaries? That is,
```julia
julia> A = Dict("a"=>1, "b"=>2); B = Dict("b"=>2, "c"=>3);

julia> merge!(+, A, B)
Dict{String,Int64} with 3 entries:
  "c" => 3
  "b" => 4
  "a" => 1
```
?

How is the combining function applied in your generator example?

**Comment by StefanKarpinski** [comment]
Yes, exactly. In the generator example, if multiple `x` have the same value of `k(x)` they would be combined, in order, pairwise, using the combiner function. This is similar to sparse matrix construction, where `+` is the default combiner.

**Comment by laborg** [comment]
Can be closed - `mergewith!` hase been implemented in 1.5.
